ui/nuxeo-page-provider-display-behavior.js

```diff
@@ -302,39 +328,60 @@ export const PageProviderDisplayBehavior = [
     },
 
     deselectItem(item) {
-      if (this.selectionEnabled) {
+      if (this.selectionEnabled && !this.selectAllActive) {
         this.$.list.deselectItem(item);
         this._updateFlags();
       }
     },
 
     deselectIndex(index) {
-      if (this.selectionEnabled) {
+      if (this.selectionEnabled && !this.selectAllActive) {
         this.$.list.deselectIndex(index);
         this._updateFlags();
       }
     },
 
     selectAll() {
-      this.items.forEach(
-        function(item) {
-          this.selectItem(item);
-        }.bind(this.$.list),
-      );
-      this._updateFlags();
+      if (this.selectionEnabled && this.selectAllEnabled) {
+        this._isSelectAllActive = true;
+        // select the visible items first to speed up the first paint (the others can be deferred)
+        const { start, end } = this._getSelectionBoundaries();
+        this._updateSelectedItems((index) => {
+          this.selectItem(this.items[index]);
+        });
+
+        // basic method to push items to selected items without proper selection, to speed up rendering
+        const pushSelectedItems = (indexStart, limit) => {
+          for (let index = indexStart; index < limit; index++) {
+            this.selectedItems.push(this.items[index]);
+          }
+        };
+        // push the items before the range
+        afterNextRender(this, pushSelectedItems, [0, start]);
+        // push the items after the range
+        afterNextRender(this, pushSelectedItems, [end + 1, this.items.length]);
+      }
     },
 
     clearSelection() {
       this.$.list.clearSelection();
+      this._isSelectAllActive = false;
       this._updateFlags();
     },
 
+    /**
+     * `true` if select all is enabled and all items are checked
+     */
+    _computeSelectAllStatus() {
+      return this.selectAllEnabled && this._isSelectAllActive;
+    },
+
     _isSelected(item) {
       return !!(this.selectedItems && this.selectedItems.length && this.selectedItems.indexOf(item) > -1);
     },
 
     _toggleSelectAll() {
-      if (this._isSelectAllChecked) {
+      if (this._isSelectAllActive) {
```

Comment:
Shouldn't this be
```suggestion
      if (this.selectAllActive) {
```
?

Reply:
Same as above.